sklearn/utils/tests/test_utils.py

```diff
@@ -209,6 +209,7 @@ def test_column_or_1d():
      (np.array([0, 1, 2], dtype=np.int64), int, True),
      (np.array([0, 1, 2], dtype=np.uint8), int, False),
      ([True, False], bool, True),
+     ([True, False], int, False),
```

Comment:
This case is to test a regression bug. Since `bool` is subclassing `int` `_check_key_type` would return the `True` for this case